.github/workflows/build_symengine.yml

```diff
@@ -0,0 +1,164 @@
+name: build symengine
+on:
+  push:
+    branches:
+      - develop
```

Comment:
Yes that is a fair point, we could run this on PRs and pushes to `main` in exactly the same way. The downside is that then for a normal release when we do a PR or push to `main` it would build the package unnecessarily (this won't cause the build to fail, however, it just won't upload anything).
Given that we do hotfixes for pytket very rarely, I think I'm happy to keep `develop` as the single place where this is tracked. If we do need to do a hotfix that changes the symengine recipe, we can do a PR to develop with the new recipe (and new version number -- builds of tket on develop will continue to use the old version), followed by the hotfix.

Reply:
Okay, I agree.